MaxText/max_utils.py

```diff
@@ -348,7 +347,10 @@ def init_initial_state(model, tx, config, is_training, key):
                           jnp.ones(input_shape, dtype=jnp.int32),
                           jnp.ones(input_shape, dtype=jnp.int32))
   if is_training:
-    return init_training_state(model.apply, model_vars['params'], tx)
+    if config.use_fp8:
+      return init_training_state(model.apply, model_vars, tx)
+    else:
+      return init_training_state(model.apply, model_vars['params'], tx)
```

Comment:
Ah is this because the flax linen impl adds those variables in the `overwrite_with_gradients` namespace?

Reply:
Yes.